app/src/main/java/com/vero/tasky/agenda/data/repository/EventRepositoryImpl.kt

```diff
@@ -0,0 +1,162 @@
+package com.vero.tasky.agenda.data.repository
+
+import com.vero.tasky.agenda.data.local.dao.EventDao
+import com.vero.tasky.agenda.data.local.dao.ModifiedAgendaItemDao
+import com.vero.tasky.agenda.data.local.entities.ModifiedAgendaItemEntity
+import com.vero.tasky.agenda.data.mappers.*
+import com.vero.tasky.agenda.data.remote.network.api.EventApi
+import com.vero.tasky.agenda.data.remote.network.dto.toAttendeeBaseInfo
+import com.vero.tasky.agenda.data.util.multipart.MultipartParser
+import com.vero.tasky.agenda.domain.model.*
+import com.vero.tasky.agenda.domain.repository.EventRepository
+import com.vero.tasky.core.data.remote.safeSuspendCall
+import kotlinx.coroutines.*
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+
+class EventRepositoryImpl(
+    private val api: EventApi,
+    private val eventDao: EventDao,
+    private val modifiedAgendaItemDao: ModifiedAgendaItemDao,
+    private val multipartParser: MultipartParser
+) : EventRepository {
+
+    override suspend fun createEvent(
+        event: AgendaItem.Event,
+    ): Result<AgendaItemUploadResult> {
+        val localPhotos = event.photos.filterIsInstance<AgendaPhoto.LocalPhoto>()
+
+        saveEvent(event = event, localPhotos = localPhotos)
+
+        var skippedPhoto = 0
+
+        safeSuspendCall {
+            withContext(Dispatchers.IO) {
+                val multipartPhotos = multipartParser.getMultipartPhotos(localPhotos)
+                skippedPhoto = localPhotos.size - multipartPhotos.size
+                api.createEvent(
+                    id = multipartParser.getMultipartId(event.id),
+                    title = multipartParser.getMultipartTitle(event.title),
+                    description = multipartParser.getMultipartDescription(event.description),
+                    from = multipartParser.getMultipartFrom(event.time),
+                    to = multipartParser.getMultipartTo(event.to),
+                    remindAt = multipartParser.getMultipartRemindAt(event.remindAt),
+                    attendeeIds = multipartParser.getMultipartAttendeesIds(event.attendees),
+                    photos = multipartPhotos
+                )
```

Comment:
It also makes sense to think about doing this with WorkManager, since uploading 10 photos could take a moment with unstable internet connection. WorkManager guarantees this will happen at some point while your upload would be cancelled when the user closes the app before it is finished